### PR TITLE
[faster-rcnn] add support for proposal

### PR DESCRIPTION
**Pull request by SeaOfOcean**
## What changes were proposed in this pull request?

proposal, anchorutil, bboxUtil and minor modification for nms
## How was this patch tested?

unit test

### CODE REVIEW

File: dl/src/main/scala/com/intel/analytics/bigdl/nn/Nms.scala [hunk 1]
```diff
@@ -74,7 +75,17 @@ class Nms extends Serializable {
     val rowLength = boxes.stride(1)
     getAreas(boxArray, offset, rowLength, boxes.size(1), areas)
     // indices start from 0
-    val orderLength = getSortedScoreInds(scores, sortIndBuffer)
+    getSortedScoreInds(scores, sortIndBuffer)
```
Comment (yiheng) [inline dl/src/main/scala/com/intel/analytics/bigdl/nn/Nms.scala]: why call it twice for sorted=false case?

File: dl/src/main/scala/com/intel/analytics/bigdl/nn/Nms.scala [hunk 2]
```diff
@@ -149,3 +162,7 @@ class Nms extends Serializable {
     overlap / ((areas(ind2) + areas(ind)) - overlap) > thresh
   }
 }
+
+object Nms {
```
Comment (yiheng) [inline dl/src/main/scala/com/intel/analytics/bigdl/nn/Nms.scala]: Does Nms need to be a class?

File: dl/src/main/scala/com/intel/analytics/bigdl/models/fasterrcnn/AnchorUtil.scala [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+package com.intel.analytics.bigdl.models.fasterrcnn
+
+import com.intel.analytics.bigdl.numeric.NumericFloat
+import com.intel.analytics.bigdl.tensor.{Storage, Tensor}
+
+/**
+ * Generates a regular grid of multi-scale, multi-aspect anchor boxes.
+ * @param param (anchor ratios, anchor scales)
+ */
+class AnchorUtil(param: AnchorParam) extends Serializable {
```
Comment (yiheng) [inline dl/src/main/scala/com/intel/analytics/bigdl/models/fasterrcnn/AnchorUtil.scala]: Can it be an object?

File: dl/src/main/scala/com/intel/analytics/bigdl/nn/Nms.scala (hunk 1 above)
Comment (SeaOfOcean) [inline dl/src/main/scala/com/intel/analytics/bigdl/nn/Nms.scala]: this is a mistake, I need to remove the first call

File: dl/src/main/scala/com/intel/analytics/bigdl/models/fasterrcnn/AnchorUtil.scala (hunk 3 above)
Comment (yiheng) [inline dl/src/main/scala/com/intel/analytics/bigdl/models/fasterrcnn/AnchorUtil.scala]: Can it be stateless impl?

Comment (yiheng) [inline dl/src/main/scala/com/intel/analytics/bigdl/models/fasterrcnn/AnchorUtil.scala]: change name to Anchor

File: dl/src/main/scala/com/intel/analytics/bigdl/models/fasterrcnn/BboxUtil.scala [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.models.fasterrcnn
+
+import com.intel.analytics.bigdl.tensor.Tensor
+
+class BboxUtil extends Serializable {
```
Comment (yiheng) [inline dl/src/main/scala/com/intel/analytics/bigdl/models/fasterrcnn/BboxUtil.scala]: change to object

File: dl/src/main/scala/com/intel/analytics/bigdl/models/fasterrcnn/AnchorUtil.scala [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+      hs = Tensor()
+    }
+    val (width, height, xCtr, yCtr) = getBasicAchorInfo(anchor)
+    ws.resizeAs(scales).mul(scales, width)
+    hs.resizeAs(scales).mul(scales, height)
+    mkAnchors(ws, hs, xCtr, yCtr)
+  }
+}
+
+object AnchorUtil {
```
Comment (yiheng) [inline dl/src/main/scala/com/intel/analytics/bigdl/models/fasterrcnn/AnchorUtil.scala]: change the name and filename

**Comment by SeaOfOcean** [comment]
pr validate pass

**Comment by yiheng** [comment]
LGTM

**Pull request closed without merging**